### PR TITLE
Adds four new butt related traits.

### PR DESCRIPTION
**Pull request by GuyonBroadway**
:cl: GuyonBroadway
add: Added the iron butt trait, you are less likley to lose your butt when farting.
add: Added the volatile butt trait, your butt is more likely to go supernova or fly off into the galaxy.
add: Added the Involuntary flatulence trait, you will fart at random intervals and occasionally superfart.
add: Added the Small anal cavity trait, your butt has one less inventory slot than normal.
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)

### CODE REVIEW

**Comment by Manly-Dwarf** [comment]
fart

**Comment by DaydreamPuppeh** [comment]
Weren't traits disabled? +1

**Comment by ghost** [comment]
not disabled, modularised and running our own trait system.

File: hippiestation/code/datums/traits/negative.dm [hunk 1]
```diff
@@ -1,2 +1,47 @@
...
-/* Currently, there are none */
\ No newline at end of file
+
+/datum/trait/flatulence
+	name = "Involuntary flatulence"
+	desc = "A spasam in the patients sphincter will cause them to uncontrollably fart at random intervals."
+	value = -1
+	gain_text = "<span class='danger'>Your butt starts to twitch!</span>"
+	lose_text = "<span class='notice'>Your butt settles down.</span>"
+	medical_record_text = "Patient has a muscular spasam in their rectal sphincter, gaseous discharge may occour."
```
Comment (JohnGinnane) [inline hippiestation/code/datums/traits/negative.dm]: "spasm"

File: hippiestation/code/datums/traits/good.dm [hunk 2]
```diff
@@ -1,2 +1,34 @@
 /* Hippie good traits */
-/* Currently, there are none */
\ No newline at end of file
+/datum/trait/iron_butt
+	name = "Iron Butt"
+	desc = "Your butt is stronger than other butts, it will be half as likley to come off when farting."
```
Comment (JohnGinnane) [inline hippiestation/code/datums/traits/good.dm]: "likely"

File: hippiestation/code/datums/traits/negative.dm [hunk 3]
```diff
@@ -1,2 +1,47 @@
...
+
+/datum/trait/flatulence/on_process()
+	if(prob(3))
+		var/obj/item/organ/butt/B = trait_holder.getorgan(/obj/item/organ/butt)
+		if(!B)
+			to_chat(trait_holder, "<span class='warning'>The building pressure in your colon hurts!</span>")
+			trait_holder.adjustBruteLoss(rand(2,6))
```
Comment (JohnGinnane) [inline hippiestation/code/datums/traits/negative.dm:16]: ouch

**Review by JohnGinnane — CHANGES_REQUESTED** [review]
just some minor typos

File: hippiestation/code/datums/traits/negative.dm [hunk 4]
```diff
@@ -1,2 +1,47 @@
...
+	value = -1
+	gain_text = "<span class='danger'>Your butt tenses up!</span>"
+	lose_text = "<span class='notice'>Your butt muscles relax.</span>"
+	medical_record_text = "Tension in the patients butt muscles has caused their anal cavity to become small."
+
+/datum/trait/smallbutt/add()
+	var/obj/item/organ/butt/B = trait_holder.getorgan(/obj/item/organ/butt)
+	if(!B)
+		to_chat(trait_holder, "<span class='warning'>You somehow gained this trait without a butt, contact an admin.</span>")
+		src.remove()
```
Comment (ghost) [inline hippiestation/code/datums/traits/negative.dm]: implicit src

File: hippiestation/code/datums/traits/negative.dm [hunk 5]
```diff
@@ -1,2 +1,47 @@
...
+	else if(B.storage_slots > 0)
+		B.inv.storage_slots = initial(B.inv.storage_slots) - 1
+	else
+		to_chat(trait_holder, "<span class='warning'>Dat booty can't get any smaller!</span>")
+		src.remove()
```
Comment (ghost) [inline hippiestation/code/datums/traits/negative.dm]: implicit src

**Review by ghost — CHANGES_REQUESTED** [review]
carbon hate the implicit src

File: hippiestation/code/datums/traits/negative.dm (hunk 4 above)
Comment (JohnGinnane) [reply hippiestation/code/datums/traits/negative.dm]: we should get an angry pepe frog, make it italian and have a speech bubble "src reeeeeeee"

**Comment by JohnGinnane** [comment]
@GuyonBroadway didn't you mention about putting the fart sounds onto the butts?

**Comment by GuyonBroadway** [comment]
Gon do dat in another PR. Atomicness and whatnot

**Comment by JohnGinnane** [comment]
cool but also good

File: hippiestation/code/datums/traits/negative.dm (hunk 5 above)
Comment (GuyonBroadway) [reply hippiestation/code/datums/traits/negative.dm]: its how tg does things to directly remove traits. 
https://github.com/HippieStation/HippieStation/blob/master/code/datums/traits/negative.dm#L59

Comment (ghost) [reply hippiestation/code/datums/traits/negative.dm]: those aren't functionally the same are they?

Comment (JohnGinnane) [reply hippiestation/code/datums/traits/negative.dm]: I think they actually do qdel(src):

https://github.com/HippieStation/HippieStation/blob/master/code/datums/traits/negative.dm#L59

Comment (ghost) [reply hippiestation/code/datums/traits/negative.dm]: further to add: remove() is called during Destroy() which i think qdel() calls
https://github.com/HippieStation/HippieStation/blob/master/code/datums/traits/_trait.dm#L32

**Comment by Carbonhell** [comment]
>butt in the pr title
automatic +1

**Comment by ghost** [comment]
gut +1

**Comment by earthmoonjune** [comment]
aye +1

File: hippiestation/code/datums/traits/negative.dm [hunk 6]
```diff
@@ -1,2 +1,47 @@
 /* Hippie Bad Traits */
-/* Currently, there are none */
\ No newline at end of file
+
+/datum/trait/flatulence
+	name = "Involuntary flatulence"
+	desc = "A spasam in the patients sphincter will cause them to uncontrollably fart at random intervals."
```
Comment (JohnGinnane) [inline hippiestation/code/datums/traits/negative.dm]: bruh this still a typo?

File: hippiestation/code/datums/traits/negative.dm [hunk 7]
```diff
@@ -1,2 +1,47 @@
...
+			trait_holder.adjustBruteLoss(rand(2,6))
+		else if(prob(1))
+			trait_holder.emote("superfart")
+		else
+			trait_holder.emote("fart")
+
+
+/datum/trait/smallbutt
+	name = "Small anal cavity"
+	desc = "Muscular contractions cause the patients anal cavity to be undersized."
```
Comment (JohnGinnane) [inline hippiestation/code/datums/traits/negative.dm]: "patient's"

File: hippiestation/code/datums/traits/negative.dm (hunk 6 above)
Comment (JohnGinnane) [reply hippiestation/code/datums/traits/negative.dm]: oh there was two of them

Comment (JohnGinnane) [reply hippiestation/code/datums/traits/negative.dm]: my b

**Review by JohnGinnane — CHANGES_REQUESTED** [review]

**Comment by Pyko1** [comment]
+1

**Comment by McHatters** [comment]
+1

**Review by JohnGinnane — APPROVED** [review]

**Pull request merged**